src/nni_manager/training_service/dlts/dltsTrainingService.ts

```diff
@@ -0,0 +1,473 @@
+            uri: `${this.dltsClusterConfig.dashboard}api/clusters/${this.dltsClusterConfig.cluster}/jobs/${trialJobDetail.dltsJobId}/status`,
+            qs: {
+                email: this.dltsClusterConfig.email,
+                password: this.dltsClusterConfig.password
+            },
+            body: {
+                status: 'killing'
+            },
+            json: true
+        };
+
+        // Set trialjobDetail's early stopped field, to mark the job's cancellation source
+        trialJobDetail.isEarlyStopped = isEarlyStopped;
+
+        await new Promise((resolve, reject) => {
+            request(options, (error: Error, response: request.Response, body: any) => {
+                if (error) {
+                    reject(error);
+                } else {
+                    resolve(body);
+                }
+            });
+        });
+    }
+
+    public async setClusterMetadata(key: string, value: string): Promise<void> {
+        switch (key) {
+            case TrialConfigMetadataKey.NNI_MANAGER_IP:
+                this.nniManagerIpConfig = <NNIManagerIpConfig>JSON.parse(value);
+                break;
+
+            case TrialConfigMetadataKey.DLTS_CLUSTER_CONFIG:
+                this.dltsClusterConfig = <DLTSClusterConfig>JSON.parse(value);
+                if (!this.dltsClusterConfig.cluster) {
+                    this.dltsClusterConfig.cluster = '.default'
+                }
+                if (!this.dltsClusterConfig.email && process.env['DLWS_USER_EMAIL']) {
+                    this.dltsClusterConfig.email = process.env['DLWS_USER_EMAIL'] as string
+                }
+                if (!this.dltsClusterConfig.password && process.env['DLTS_JOB_TOKEN']) {
+                    this.dltsClusterConfig.password = process.env['DLTS_JOB_TOKEN'] as string
+                }
+                if (!this.dltsClusterConfig.team && process.env['DLWS_VC_NAME']) {
+                    this.dltsClusterConfig.team = process.env['DLWS_VC_NAME'] as string
+                }
+                const gpuRequestOptions: request.Options = {
+                    method: 'GET',
+                    qs: {
+                        email: this.dltsClusterConfig.email,
+                        password: this.dltsClusterConfig.password
+                    },
+                    uri: `${this.dltsClusterConfig.dashboard}api/teams/${this.dltsClusterConfig.team}/clusters/${this.dltsClusterConfig.cluster}`,
+                    json: true
+                };
+                const gpu = await new Promise<string>((resolve, reject) => {
+                    request(gpuRequestOptions, (error, response, data) => {
+                        if (error) {
+                            return reject(error)
+                        }
+                        let gpus = Object.keys(data['gpu_capacity'])
+                        resolve(gpus[0])
+                    })
+                });
+                this.dltsClusterConfig.gpuType = gpu;
```

Comment:
Can users set gpuType in config file?

Reply:
Yes and no since there is only one choice at present, but it have to be configured correctly or DLTS could not schedule it correctly.

DLTS plans to support cluster with multiple gpuTypes in the future.